android-o/AutofillFramework/AutofillFramework/VirtualSignInActivity.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Support.V7.App;
+using Android.Views;
+using Android.Views.Autofill;
+using Android.Widget;
+using Java.Lang;
+
+namespace AutofillFramework
+{
+    [Activity(Label = "VirtualSignInActivity")]
+    [Register("com.xamarin.AutofillFramework.VirtualSignInActivity")]
+    public class VirtualSignInActivity : AppCompatActivity
+    {
+        private CustomVirtualView mCustomVirtualView;
+        private AutofillManager mAutofillManager;
+        private CustomVirtualView.Line mUsernameLine;
+        private CustomVirtualView.Line mPasswordLine;
+
+        public static Intent GetStartActivityIntent(Context context)
+        {
+            var intent = new Intent(context, typeof(VirtualSignInActivity));
+            return intent;
+        }
+
+        protected override void OnCreate(Bundle savedInstanceState)
+        {
+            base.OnCreate(savedInstanceState);
+            SetContentView(Resource.Layout.virtual_login_activity);
+
+            mCustomVirtualView = FindViewById<CustomVirtualView>(Resource.Id.custom_view);
+
+            var credentialsPartition =
+                mCustomVirtualView.AddPartition(GetString(Resource.String.partition_credentials));
+            mUsernameLine = credentialsPartition.AddLine("username", (int) AutofillType.Text,
+                GetString(Resource.String.username_label),
+                "         ", false, View.AutofillHintUsername);
+            mPasswordLine = credentialsPartition.AddLine("password", (int) AutofillType.Text,
```

Comment:
We can remove this cast once the `AddLine` signature has been changed to accept an `AutofillType`.